scripts/test/runtests.ps1

```diff
@@ -5,65 +5,79 @@
+        if (!$?) {
+            Write-Host Command failed: dotnet publish --framework "dnxcore50" --runtime "$Rid" --output "$TestBinRoot" --configuration "$Configuration" "$RepoRoot\test\$_"
+            exit 1
+        }
     }
-}
 
-## Temporary Workaround for Native Compilation
-## Need x64 Native Tools Dev Prompt Env Vars
-## Tracked Here: https://github.com/dotnet/cli/issues/301
-pushd "$env:VS140COMNTOOLS\..\..\VC"
-cmd /c "vcvarsall.bat x64&set" |
-foreach {
-  if ($_ -match "=") {
-    $v = $_.split("=", 2); set-item -force -literalpath "ENV:\$($v[0])" -value "$($v[1])"
-  }
-}
-popd
+    ## Temporary Workaround for Native Compilation
+    ## Need x64 Native Tools Dev Prompt Env Vars
+    ## Tracked Here: https://github.com/dotnet/cli/issues/301
+    pushd "$env:VS140COMNTOOLS\..\..\VC"
+    cmd /c "vcvarsall.bat x64&set" |
+    foreach {
+      if ($_ -match "=") {
+        $v = $_.split("=", 2); set-item -force -literalpath "ENV:\$($v[0])" -value "$($v[1])"
+      }
+    }
+    popd
 
-# copy TestProjects folder which is used by the test cases
-mkdir -Force "$TestBinRoot\TestProjects"
-cp -rec -Force "$RepoRoot\test\TestProjects\*" "$TestBinRoot\TestProjects"
+    # copy TestProjects folder which is used by the test cases
+    mkdir -Force "$TestBinRoot\TestProjects"
+    cp -rec -Force "$RepoRoot\test\TestProjects\*" "$TestBinRoot\TestProjects"
 
-$failCount = 0
-$failingTests = @()
+    $failCount = 0
+    $failingTests = @()
 
-pushd "$TestBinRoot"
+    pushd "$TestBinRoot"
 
-# Run each test project
-$TestProjects | ForEach-Object {
-    & "corerun.exe"  "xunit.console.netcore.exe" "$_.dll" -xml "$_-testResults.xml" -notrait category=failing
-    $exitCode = $LastExitCode
-    if ($exitCode -ne 0) {
-        $failingTests += "$_"
-    }
+    # Run each test project
+    $TestProjects | ForEach-Object {
+        # corehost is on the PATH because it's part of the stage 2 build
+        & "corehost" "xunit.console.netcore.exe" "$_.dll" -xml "$_-testResults.xml" -notrait category=failing
+        $exitCode = $LastExitCode
+        if ($exitCode -ne 0) {
+            $failingTests += "$_"
+        }
 
-    $failCount += $exitCode
-}
+        $failCount += $exitCode
+    }
 
-popd
+    popd
 
-if ($failCount -ne 0) {
-    Write-Host -ForegroundColor Red "The following tests failed."
-    $failingTests | ForEach-Object {
-        Write-Host -ForegroundColor Red "$_.dll failed. Logs in '$TestBinRoot\$_-testResults.xml'"
+    if ($failCount -ne 0) {
+        Write-Host -ForegroundColor Red "The following tests failed."
+        $failingTests | ForEach-Object {
+            Write-Host -ForegroundColor Red "$_.dll failed. Logs in '$TestBinRoot\$_-testResults.xml'"
+        }
+    } else {
+        Write-Host -ForegroundColor Green "All the tests passed!"
     }
-}
-else {
-    Write-Host -ForegroundColor Green "All the tests passed!"
+} finally {
```

Comment:
I am not an expert in PS, so asking it here. Will this finally block run even if the user interrupts the script, say with a CTRL+C command?! Do we need to use something like this to guarantee that this will run even in the case of a CTRL+C: [console]::TreatControlCAsInput = $true?